src/masternode/masternode-sync.cpp

```diff
@@ -115,7 +115,7 @@ void CMasternodeSync::ProcessTick(CConnman& connman)
 
     // reset the sync process if the last call to this function was more than 60 minutes ago (client was in sleep mode)
     static int64_t nTimeLastProcess = GetTime();
-    if(GetTime() - nTimeLastProcess > 60*60) {
+    if(GetTime() - nTimeLastProcess > 60*60 && Params().NetworkIDString() != CBaseChainParams::REGTEST) {
```

Comment:
Would prefer using `force_finish_mnsync` in related tests instead.

Reply:
This is what I did before and I kept running into issues. For example, when bumping mock time too fast (60 mock minutes in 1 real second), this will unexpectedly cause new connections to be dropped, which in turn causes spontaneous DKG failures. And when I found such places where it happened, I'd always had to sleep for a few seconds before forcing mnsync to make sure that `ProcessTick` is actually executed before.

Hunting down these issues was a mess and so I decided to just remove this behavior from regtest (which really doesn't make sense here because regtest nodes don't really "hibernate").

Reply:
Well,regtest nodes _could_ "hibernate", we just don't test this :) Anyway, since this is an issue for regtest MNs and real MNs don't/shouldn't "hibernate", how about:
```suggestion
    if(GetTime() - nTimeLastProcess > 60*60 && !fMasternodeMode) {
```
instead?

Reply:
This wouldn't fix the issue, as it's specifically about masternodes trying to connect to other masternodes. And regtest masternodes are the same as "normal" masternodes, so `fMasternodeMode` is true for them as well.

Reply:
I don't understand... adding `!fMasternodeMode` would allow us skip this block and avoid reseting mnsync status on MNs (including cases when we bump mocktime in regtest). Isn't this what you wanted to do here?

Reply:
Ahhh...I kind of mixed up your suggestion with `fMasternodeMode==true`...you're right, using `!fMasternodeMode` will also work. I'll fix it in a few minutes.

Reply:
done